### DISCUSSION TITLE
Set-PnPListItem: The request was canceled due to the configured HttpClient.Timeout of 100 seconds elapsing

### DISCUSSION THREAD

**Discussion by gruering** [pnp/powershell#3336] [General] [answered]
Trying to update lots of items in a SharePoint list with Set-PnPListItem results in an error:
**The request was cancelled due to the configured HttpClient.Timeout of 100 seconds elapsing**

Changing the RequestTimeout property on the context object of the connection did not help.

### Expected behavior 
RequestTimeout property of the context object in the connection prevents the error

### Actual behavior 
Could not find any way around this error so far.

### Steps to reproduce behavior
Update all list items in a list containing 3000 items.

### What is the version of the Cmdlet module you are running?
2.2.0

### Which operating system/environment are you running PnP PowerShell on?
- [X] Windows

**Comment by jpwegmuller50** [comment, marked as answer]
I'm running into the same issue updating 55,000+ items.  I'm using a quick and dirty way around this by retrying one time if it times out (sample code below).  So far that's been enough of a workaround.

```
          try {
            Set-PnPListItem -List $MatrixList -Identity $matrixItemId -Values $values -UpdateType UpdateOverwriteVersion -Batch $batch
          } catch {
            # if timed out, try again
            if ($PSItem.Exception.Message.IndexOf("HttpClient.Timeout") -ne -1) {
              Write-Host "  timed out, trying again..."
              Set-PnPListItem -List $MatrixList -Identity $matrixItemId -Values $values -UpdateType UpdateOverwriteVersion -Batch $batch
            } else {
              throw $_
            }
          }
```

**Comment by Offbeatmammal** [comment]
Would be nice if https://github.com/pnp/powershell/issues/4899 allows us to override the default 100 second timeout
